src/Shape.cpp

```diff
@@ -22,6 +22,71 @@ bool Shape::intersect(const Ray &, float t0, float t1, double& distance) const {
     return false;
 }
 
+bool Shape::intersect(const Vector3& p) const {
+    if (p.length2() > mesh_.getSquaredMaxRadius()){
+        return false;
+    }
+
+    int intersect_count = 0;
+
+    // determine plucker coordinates of line p
+    Vector3 p_out = Vector3(1.1 * mesh_.getMaxRadius(), 0, 0);
+    Vector3 p_U = p-p_out;
+    Vector3 p_V = p.cross(p_out);
+
+    // load triangles
+    const std::vector<geo::Vector3>& t_points = mesh_.getPoints();
+    const std::vector<TriangleI>& triangles_i = mesh_.getTriangleIs();
+    for(std::vector<TriangleI>::const_iterator it = triangles_i.begin(); it != triangles_i.end(); ++it) {
+        Vector3 v1 = t_points[it->i1_];
+        Vector3 v2 = t_points[it->i2_];
+        Vector3 v3 = t_points[it->i3_];
+
+        Vector3 e1_U = v1-v2;
+        Vector3 e2_U = v2-v3;
+        Vector3 e3_U = v3-v1;
+
+        Vector3 e1_V = v1.cross(v2);
+        Vector3 e2_V = v2.cross(v3);
+        Vector3 e3_V = v3.cross(v1);
```

Comment:
Creating all these new instances is not efficient. Maybe @alberth could help to find the most efficient solution